File: 154. Find Minimum in Rotated Sorted Array II/154. Find Minimum in Rotated Sorted Array II.cpp
class Solution
{
public:
    int findMin(vector<int> &nums)
    {
        int n = nums.size();
        int i = 0;
        int j = n - 1;
        while (i < j)
        {
            int mid = (i + j) / 2;
            if (nums[mid] > nums[j])
            {
                i = mid + 1;
            }
            else
                j--;
        }
        return nums[i];
    }
};

File: 154. Find Minimum in Rotated Sorted Array II/154. Find Minimum in Rotated Sorted Array II.java
class Solution {
    public int findMin(int[] arr) {
       
        int i=0,j=arr.length-1;
        if(arr[i]<arr[j])
            return arr[i];
        while(i<j)
        {
            int mid=i+(j-i)/2;
            if(arr[mid]>arr[j])
                i=mid+1;
            else  if(arr[mid]<arr[j])
                j=mid;
            else 
                j--;
        }
        return arr[j];
    }
}